src/Illuminate/Http/Concerns/InteractsWithInput.php

```diff
@@ -307,6 +308,30 @@ public function collect($key = null)
         return collect(is_array($key) ? $this->only($key) : $this->input($key));
     }
 
+    /**
+     * Retrieve input from the request as a Carbon instance.
+     *
+     * @param  string  $key
+     * @param  string|null  $format
+     * @param  string|null  $tz
+     * @param  \Closure|string|null  $default
+     * @return \DateTimeInterface|\Illuminate\Support\Carbon|null
+     */
+    public function datetime($key, $format = null, $tz = null, $default = null)
+    {
+        $date = $this->input($key) ?? value($default, $this);
+
+        if (is_null($date)) {
+            return null;
+        }
+
+        if ($format) {
+            return Date::createFromFormat($format, $date, $tz);
```

Comment:
`createFromFormat()` may return `false`, which is not accounted for. Here's an example of how this is handled elsewhere in the framework:

https://github.com/laravel/framework/blob/8.x/src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php#L1141-L1147

Also note that `createFromFormat()` may throw `InvalidFormatException` (hence the `InvalidArgumentException` catch in the implementation above).

Reply:
That's assuming the developer doesn't validates an input from being a valid datetime. I think it should return `false` to keep it consistent with how it's handled on models.